### ISSUE TITLE
Resource 'data.azurerm_client_config.current' does not have attribute 'service_principal_application_id'

### ISSUE THREAD

**Issue by ahanoff** [terraform-providers/terraform-provider-azurerm#2108]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.8
provider.azurerm v1.16.0

### Affected Resource(s)

* azurerm_client_config

### Terraform Configuration Files

```hcl
data "azurerm_client_config" "current" {}

output "account_id" {
  value = "${data.azurerm_client_config.current.client_id}"
}

output "tenant_id" {
  value = "${data.azurerm_client_config.current.tenant_id}"
}

output "subscription_id" {
  value = "${data.azurerm_client_config.current.subscription_id }"
}

output "service_principal_application_id" {
  value = "${data.azurerm_client_config.current.service_principal_application_id}"
}

output "service_principal_object_id" {
  value = "${data.azurerm_client_config.current.service_principal_object_id}"
}
```

### Debug Output

https://gist.github.com/ahanoff/a90e87e18294ad1e0b0344a4b0ff75ba

### Expected Behavior

Return service_principal_object_id and service_principal_application_id as described in documentation

### Actual Behavior

Error output: Resource 'data.azurerm_client_config.current' does not have attribute 'service_principal_application_id'

### Steps to Reproduce

1. terraform init
2. terraform apply

### References

* #0000

**Comment by katbyte** [comment]
Hi @ahanoff,

How are authenticating with azure? Currently the datasource does not populate those fields if a service principle is not being used and because of that if you access them there is an error.

I have opened #2120 so that they will instead be set to an empty string.

**Comment by ahanoff** [comment]
Yes, you are right, I don't use service principal. Empty string is good,
thank you!

**Comment by tombuildsstuff** [comment]
hi @ahanoff 

Just to let you know that this has been released as a part of v1.18 of the AzureRM Provider ([the full changelog is available here](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.18.0/CHANGELOG.md)). You can upgrade to this by specifying the version in the provider block (as shown below) and then running `terraform init -upgrade`

```
provider "azurerm" {
  version = "=1.18.0"
}
```

Thanks!
